plugins/woocommerce-blocks/assets/js/hocs/with-product-variations.tsx

```diff
@@ -81,7 +108,11 @@ const withProductVariations = createHigherOrderComponent(
 
 				this.setState( { loading: true } );
 
-				getProductVariations( expandedProduct )
+				(
+					getProductVariations( expandedProduct ) as Promise<
```

Comment:
Not saying you have to do this, but one way you could avoid the cast here is to start moving these utils across to ts separately too. For example `getProductVariations` is in a `utils/index.js` file, you could separate it out to a TS file, pass `ProductResponseVariationsItem[]` to `apiFetch` generic arg and then that util would be able to be correctly typed by everyone without casting which would be preferable.

Reply:
This is a great suggestion! I tried to keep the changes inside this file because we have many issues created for each file that we need to convert to TS. I believe we can open a new issue to work on this `utils/index.js` file since the functions exported from it can also impact other files. WDYT?

Reply:
Yep I'm fine with that because this wouldn't be a big job, so happy to keep it to another PR